qiita_db/processing_job.py

```diff
@@ -128,14 +128,43 @@ def create(cls, user, parameters):
             The user executing the job
         parameters : qiita_db.software.Parameters
             The parameters of the job being executed
+        force : bool
+            Force creation on duplicated parameters
 
         Returns
         -------
         qiita_db.processing_job.ProcessingJob
             The newly created job
+
+        Notes
+        -----
+        If force is True the job is gonna be created even if another job
+        exists with the same parameters
         """
         with qdb.sql_connection.TRN:
             command = parameters.command
+
+            # check if a job with the same parameters already exists
+            sql = """SELECT processing_job_id, email, processing_job_status
+                     FROM qiita.processing_job
+                     LEFT JOIN qiita.processing_job_status
+                        USING (processing_job_status_id)
+                     WHERE command_id = %s AND processing_job_status IN (
+                        'success', 'waiting', 'running') """
+            params = " AND ".join(
+                ["command_parameters->>'%s' = '%s'" % (k, v)
```

Comment:
This is one confusing double conditional list comprehension. `for v in v`?

Reply:
agree